src/main/java/com/github/pedrovgs/androidwifiadb/adb/AndroidWiFiADB.java

```diff
@@ -0,0 +1,161 @@
+import com.github.pedrovgs.androidwifiadb.view.View;
+import com.github.pedrovgs.androidwifiadb.model.Device;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AndroidWiFiADB {
+
+  private final ADB adb;
+  private final View view;
+  private List<Device> devices;
+
+  public AndroidWiFiADB(ADB adb, View view) {
+    this.adb = adb;
+    this.view = view;
+    this.devices = new ArrayList<Device>();
+  }
+
+  public void connectDevices() {
+    if (!isADBInstalled()) {
+      view.showADBNotInstalledNotification();
+      return;
+    }
+    devices = adb.getDevicesConnectedByUSB();
+    if (devices.isEmpty()) {
+      view.showNoConnectedDevicesNotification();
+      return;
+    }
+
+    devices = adb.connectDevices(devices);
+    showConnectionResultNotification(devices);
+  }
+
+  /**
+   * Find connected devices through USB.
+   * If there's a new device - we add it to devices list.
+   * Call <code>getDevices()</code> in order to get updated devices list.
+   * <br>We should always update devices list,
+   * because their names might change.
+   * Unathorized and authorized states produce different device names.
+   * @return true - refresh required, false - otherwise.
+   */
+  public boolean refreshDevicesList() {
+    if (!isADBInstalled()) {
+      return false;
+    }
+    List<Device> connected = adb.getDevicesConnectedByUSB();
+    for (Device connectedDevice : connected) {
+      boolean deviceExists = checkDeviceExistance(connectedDevice);
+      if (!deviceExists) {
+        connectedDevice.setIp(adb.getDeviceIp(connectedDevice));
+        devices.add(connectedDevice);
+      } else {
+        updateDeviceInfo(connectedDevice);
+      }
+    }
+    return true;
+  }
+
+  private void updateDeviceInfo(Device updatedDevice) {
+    for (int i = 0, size = devices.size(); i < size; i++) {
+      Device device = devices.get(i);
+      if (updatedDevice.getId().equals(device.getId())) {
+        devices.remove(i);
+        device.setName(updatedDevice.getName());
+        devices.add(i, device);
+      }
+    }
+  }
+
+  private boolean checkDeviceExistance(Device connectedDevice) {
+    boolean deviceExists = false;
+    for (Device device : devices) {
+      if (connectedDevice.getId().equals(device.getId())) {
+        deviceExists = true;
+      }
+    }
+    return deviceExists;
+  }
+
+  public List<Device> getDevices() {
+    return devices;
+  }
+
+  private boolean isADBInstalled() {
+    return adb.isInstalled();
+  }
+
+  public void updateProject(Project project) {
+    this.adb.updateProject(project);
+  }
+
+  public void connectDevice(Device device) {
+    if (!isADBInstalled()) {
+      view.showADBNotInstalledNotification();
+      return;
+    }
+
+    List<Device> connectedDevices = new ArrayList<Device>();
+    connectedDevices.add(device);
+    connectedDevices = adb.connectDevices(connectedDevices);
+    for (Device connected : connectedDevices) {
+      updateDeviceInfo(connected);
```

Comment:
You are updating the connected device state so we could improve the rename this method to ``updateDeviceConnectionState``.